march_data_collector/src/march_data_collector/esp_adapter.py

```diff
@@ -339,7 +361,7 @@ def get_join_time_str(timestamp, frequency):
 
     :param data: ROS timestamp message std_msgs/stamp
     """
-    time = round(timestamp.secs + timestamp.nsecs * 10 ** (-9) * frequency) / frequency
+    time = round((timestamp.secs + timestamp.nsecs * 10 ** (-9)) * frequency) / frequency
```

Comment:
I think this should also work, not extremely relevant but a nice touch maybe. 
```suggestion
    time = round(timestamp.to_sec() * frequency) / frequency
```

Reply:
@RutgerVanBeek are you implementing this change or not? Since you gave a :+1:

Reply:
Yes I am planning on doing this today.